spec/operators/groupBy-spec.ts

```diff
@@ -903,6 +903,51 @@ describe('Observable.prototype.groupBy', () => {
     expectObservable(source, unsub).toBe(expected, expectedGroups);
   });
 
+  it('should dispose a durationSelector after closing the group',
+  () => {
+    let durationMarble = '-------s';
+    let durations = [];
+    const duration = () => {
+      let o = cold(durationMarble); // duration 7 frames, then stop
+      durations.push(o);
+      return o;
+    };
+    const e1 = hot( '-1-2-1-3-1-4-1-5------------|');
+    const sub =     '^                            !' ;
+    const gr =      '-a-k---l-b-m---n------------|' ;
+    const empty =   '-----------------------------' ;
+
+    let unsubs = [];
+    for (let i = 0; i < gr.length; i++) {
+      if (gr[i] !== '-' && gr[i] !== '|') {
+        unsubs.push(empty.slice(0, i) + '^' + durationMarble.slice(1, -1) + '!');
```

Comment:
This is arguably less readable than just using the strings:

```
const unsubs = [
  '^------!',
  '-^------!',
  '--^------!,
 /* etc */
];
```

It's more busy work, sure, but tests are about readability.